pxd.c

```diff
@@ -631,6 +631,104 @@ static inline unsigned int get_op_flags(struct bio *bio)
 	return op_flags;
 }
 
+static
+void pxd_process_ioswitch_complete(struct fuse_conn *fc, struct fuse_req *req,
+	int status)
+{
+	struct pxd_device *pxd_dev = req->pxd_dev;
+
+	/// io path switch event completes with status.
+	printk("device %llu completed ioswitch %d with status %d\n",
+		pxd_dev->dev_id, req->in.opcode, status);
+
+	if (req->in.opcode == PXD_FAILOVER) {
+		// if the status is successful, then reissue IO to userspace
+		// else fail IO to complete.
+		spin_lock(&pxd_dev->fp.fail_lock);
+		__pxd_reissuefailQ(pxd_dev, status);
+		pxd_dev->fp.active_failover = PXD_FP_FAILOVER_NONE;
+		spin_unlock(&pxd_dev->fp.fail_lock);
+	}
+
+	// reopen the suspended device
```

Comment:
further down... currently IOs that have errored in fastpath try to failover, and based on the status are reissued to userspace or immediately failed. I should probably extend the abort IO path on timer expiry to cleanup these IO... will revise.

Reply:
fastpath IO tries to initiate failover to userspace on any IO failure. If this fails, then IO is failed to userspace immediately.
failover initiation is through an inline IO request through the fuse path. So it can be enqueued even when px is down.